pkg/tasks/prometheusoperator.go

```diff
@@ -84,7 +84,12 @@ func (t *PrometheusOperatorTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling Prometheus Operator Service failed")
 	}
 
-	d, err := t.factory.PrometheusOperatorDeployment()
+	config, err := t.client.GetAPIServerConfig(ctx, "cluster")
+	if err != nil {
+		return err
```

Comment:
maybe wrap the error message to be consistent with the rest?

```suggestion
		return errors.Wrap(err, "failed to get API server configuration")
```